src/anomalib/models/efficientad/lightning_model.py

```diff
@@ -0,0 +1,248 @@
+"""EfficientAD: Accurate Visual Anomaly Detection at Millisecond-Level Latencies.
+https://arxiv.org/pdf/2303.14535.pdf
+"""
+
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import logging
+from pathlib import Path
+
+import albumentations as A
+import numpy as np
+import torch
+import tqdm
+from albumentations.pytorch import ToTensorV2
+from omegaconf import DictConfig, ListConfig
+from pytorch_lightning.utilities.types import STEP_OUTPUT
+from torch import Tensor, optim
+from torch.utils.data import DataLoader
+from torchvision.datasets import ImageFolder
+
+from anomalib.data.utils import DownloadInfo, download_and_extract
+from anomalib.models.components import AnomalyModule
+
+from .torch_model import EfficientADModel
+
+logger = logging.getLogger(__name__)
+
+IMAGENETTE_DOWNLOAD_INFO = DownloadInfo(
+    name="imagenette2.tgz",
+    url="https://s3.amazonaws.com/fast-ai-imageclas/imagenette2.tgz",
+    hash="fe2fc210e6bb7c5664d602c3cd71e612",
+)
+
+
+class TransformsWrapper:
+    def __init__(self, t: A.Compose):
+        self.transforms = t
+
+    def __call__(self, img, *args, **kwargs):
+        return self.transforms(image=np.array(img))
+
+
+class EfficientAD(AnomalyModule):
+    """PL Lightning Module for the EfficientAD algorithm."""
+
+    def __init__(
+        self,
+        category_name: str,
+        teacher_file_name: str,
+        teacher_out_channels: int,
+        pre_trained_dir: str,
+        image_size: list,
+        model_size: str = "M",
```

Comment:
Should we make this an Enum? It will be easier to see the available options then.